### PR TITLE
[pylint] Updating PylintCheckers testing

### LINKED ISSUE #23382: [pylint] Add test for PackageNameDoesNotUseUnderscoreOrPeriod,

### PR DESCRIPTION
**Pull request by l0lawrence**
- CheckNamingMismatchGeneratedCode: Adding guidelines link test, and updating naming convention
- CheckEnum: Adding guidelines link test for both links (using helper function)
- CheckAPIVersion: Adding guidelines link test 
- FileHasCopyrightHeader: Added in acceptable and violation test files and accompanying tests
- TestPackageNameDoesNotUseUnderscoreOrPeriod: Added in a testing class for acceptable and violation package name
and added in guidelines link test
- TestServiceClientUsesNameWithClientSuffix: added in this testing class with guidelines link test checker

fixes #23382

### CODE REVIEW

File: scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py [hunk 1]
```diff
@@ -2784,6 +2882,19 @@ def test_enum_file_both_violation(self):
...
+        assert response_one.http_response.status_code == 200
+        response_two = self._create_url_pipeline("https://azure.github.io/azure-sdk/python_implementation.html#extensible-enumerations")
+        assert response_two.http_response.status_code == 200
+    
+    def _create_url_pipeline(self,url):
+        config = Configuration()
+        client = PipelineClient(url, config=config)
+        request = client.get(url)
+        response = client._pipeline.run(request)
+        return response
```
Comment (kristapratico) [inline scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py:2896]: we can simplify this and just use `requests` here

```
import requests

resp = requests.get("url")
assert resp.status_code == 200
```

File: scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py [hunk 2]
```diff
@@ -2678,6 +2768,14 @@ def test_disable_pylint_alias(self):
         with self.assertNoMessages():
             self.checker.visit_module(node)
 
+    def test_guidelines_link_active(self):
+        url = "https://github.com/Azure/autorest/blob/main/docs/generate/built-in-directives.md"
```
Comment (kristapratico) [inline scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py:2772]: do we actually include this link in the linting error?

**Review by kristapratico — APPROVED** [review]

File: scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py (hunk 2 above)
Comment (l0lawrence) [reply scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py:2772]: Nope, I will add it!

Comment (l0lawrence) [reply scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py:2772]: C4745: Do not alias generated code. This messes up sphinx, intellisense, and apiview, so please modify the name of the generated code through the swagger / directives, or code customizations. See Details: https://github.com/Azure/autorest/blob/main/docs/generate/built-in-directives.md (naming-mismatch)

**Pull request merged**